credsweeper/utils/util.py

```diff
@@ -270,6 +272,53 @@ def read_data(path: str) -> Optional[bytes]:
             logger.error(f"Unexpected Error: Can not read '{path}'. Error message: '{exc}'")
         return None
 
+    @staticmethod
+    def get_xml_data(file_path: str) -> Tuple[Optional[List[str]], Optional[List[int]]]:
+        """Read xml data and return List of str.
+
+        Try to read the xml data and return formatted string.
+
+        Args:
+            file_path: path of xml file
+
+        Return:
+            List of formatted string(f"{root.tag} : {root.text}")
+
+        """
+        lines = []
+        line_nums = []
+        try:
+            with open(file_path, "r") as f:
+                xml_lines = f.readlines()
+            tree = etree.fromstringlist(xml_lines)
+            for element in tree.iter():
+                tag = Util._extract_element_data(element, "tag")
+                text = Util._extract_element_data(element, "text")
+                lines.append(f"{tag} : {text}")
+                line_nums.append(element.sourceline)
+        except Exception as exc:
+            logger.error(f"Cannot parse '{file_path}' to xml {exc}")
+            return None, None
+        return lines, line_nums
+
+    @staticmethod
+    def _extract_element_data(element, attr) -> str:
+        """Extract xml element data to string.
+
+        Try to extract the xml data and strip() the string.
+
+        Args:
+            element: xml element
+            attr: attribute name
+
+        Return:
+            String xml data with strip()
+
+        """
+        if getattr(element, attr) is None or not isinstance(getattr(element, attr), str):
+            return ""
+        return str(getattr(element, attr)).strip()
```

Comment:
```suggestion
        element_attr = getattr(element, attr)
        if element_attr is None or not isinstance(element_attr, str):
            return ""
        return element_attr.strip()
```
a bit optimization(?) not sure whether python does it

Reply:
Thank you for suggestion. I refactored the code with f00c5e1.
Please check.